keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,295 @@
+
+    This layer can be thought of as an all in one image augmentation layer.  The policy
+    implemented by this layer has been benchmarked extensively and is effective on a
+    wide variety of datasets.
+
+    The policy operates as follows:
+
+    For each `layer`, the policy selects a random operation from a list of operations.
+    It then samples a random number and if that number is less than
+    `rate` applies it to the given image.
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        augmentations_per_image: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.  magnitude should
+            be a float in the range `[0, 1]`.  A magnitude of `0` indicates that the
+            augmentations are as weak as possible (not recommended), while a value of
+            `1.0` implies use of the strongest possible augmentation.  Defaults to
+            `0.5`.
+        magnitude_stddev: the standard deviation to use when drawing values
+            for the perturbations.  Keep in mind magnitude will still be clipped to the
+            range `[0, 1]` after samples are drawn from the uniform distribution.
+            Defaults to `0.15`.
+        rate:  the rate at which to apply each augmentation.  This parameter is applied
+            on a per-distortion layer, per image.
+
+    Usage:
+    ```python
+    (x_test, y_test), _ = tf.keras.datasets.cifar10.load_data()
+    rand_augment = keras_cv.layers.RandAugment(
+        value_range=(0, 255), augmentations_per_image=3, magnitude=5.0
+    )
+    x_test = rand_augment(x_test)
+    ```
+    """
+
+    def __init__(
+        self,
+        value_range,
+        augmentations_per_image=3,
+        magnitude=0.5,
+        magnitude_stddev=0.15,
+        rate=1.0,
+        seed=None,
+        **kwargs,
+    ):
+        # As an optimization RandAugment makes all internal layers use (0, 255) while
+        # and we handle range transformation at the _augment level.
+        super().__init__(
+            layers=RandAugment.get_standard_policy(
+                (0, 255), magnitude, magnitude_stddev, seed=seed
+            ),
+            augmentations_per_image=augmentations_per_image,
+            rate=rate,
+            **kwargs,
+            seed=seed,
+        )
+        self.magnitude = float(magnitude)
+        self.value_range = value_range
+        self.seed = seed
+        if magnitude < 0.0 or magnitude > 1:
+            raise ValueError(
+                f"`magnitude` must be in the range [0, 1], got `magnitude={magnitude}`"
+            )
+        self.magnitude_stddev = float(magnitude_stddev)
```

Comment:
Should we also do any validation for param like magnitude_stddev as well?

Reply:
Done.